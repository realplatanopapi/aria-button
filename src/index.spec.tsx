import React from 'react'
import {shallow} from 'enzyme'
import {spy} from 'sinon'

import AriaButton from '.'

const createKeyPressEvent = (key: string) => ({
  preventDefault: spy(),
  key
})

test('renders children in a span when no component is specified', () => {
  const wrapper = shallow(
    <AriaButton>Click me</AriaButton>
  )
  expect(wrapper).toMatchSnapshot()
})

test('renders children in custom component', () => {
  const wrapper = shallow(
    <AriaButton tag="div">Click me</AriaButton>
  )
  expect(wrapper).toMatchSnapshot()
})

test('invokes onClick when clicked', () => {
  const onClick = spy()
  const wrapper = shallow(
    <AriaButton onClick={onClick}>Click me</AriaButton>
  )
  wrapper.simulate('click')
  expect(onClick.callCount).toBe(1)
})

test('invokes onClick when the Space key is pressed', () => {
  const onClick = spy()
  const wrapper = shallow(
    <AriaButton onClick={onClick}>Click me</AriaButton>
  )
  const keyPressEvent = createKeyPressEvent(' ')
  wrapper.simulate('keypress', keyPressEvent)
  expect(onClick.callCount).toBe(1)
  expect(keyPressEvent.preventDefault.callCount).toBe(1)
})

test('invokes onClick when the Enter key is pressed', () => {
  const onClick = spy()
  const wrapper = shallow(
    <AriaButton onClick={onClick}>Click me</AriaButton>
  )
  const keyPressEvent = createKeyPressEvent(' ')
  wrapper.simulate('keypress', keyPressEvent)
  expect(onClick.callCount).toBe(1)
  expect(keyPressEvent.preventDefault.callCount).toBe(1)
})

test('invokes onKeyPress when onClick is fired', () => {
  const onClick = spy()
  const onKeyPress = spy()
  const wrapper = shallow(
    <AriaButton onClick={onClick} onKeyPress={onKeyPress}>Click me</AriaButton>
  )
  wrapper.simulate('keypress', createKeyPressEvent('Enter'))
  expect(onKeyPress.callCount).toBe(1)
  expect(onClick.callCount).toBe(1)
})