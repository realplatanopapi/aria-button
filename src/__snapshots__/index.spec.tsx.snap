// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children in a span when no tag is specified 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AriaButton>
    Click me
  </AriaButton>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Click me",
      "onClick": undefined,
      "onKeyPress": [Function],
      "role": "button",
      "tabIndex": 0,
    },
    "ref": null,
    "rendered": "Click me",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Click me",
        "onClick": undefined,
        "onKeyPress": [Function],
        "role": "button",
        "tabIndex": 0,
      },
      "ref": null,
      "rendered": "Click me",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders children in custom tag 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AriaButton
    tag="div"
  >
    Click me
  </AriaButton>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Click me",
      "onClick": undefined,
      "onKeyPress": [Function],
      "role": "button",
      "tabIndex": 0,
    },
    "ref": null,
    "rendered": "Click me",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Click me",
        "onClick": undefined,
        "onKeyPress": [Function],
        "role": "button",
        "tabIndex": 0,
      },
      "ref": null,
      "rendered": "Click me",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
